src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -0,0 +1,296 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Security;
+using System.Security.Principal;
+using System.Threading;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Authentication.ExtendedProtection;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Security
+{
+    //
+    // The class maintains the state of the authentication process and the security context.
+    // It encapsulates security context and does the real work in authentication and
+    // user data encryption with NEGO SSPI package.
+    //
+    // This is part of the NegotiateStream PAL.
+    //
+    internal static class NegotiateStreamPal
+    {
+        // value should match the Windows sspicli NTE_FAIL value
+        // defined in winerror.h
+        private const int NTE_FAIL = unchecked((int)0x80090020);
+
+        internal static IIdentity GetIdentity(NTAuthentication context)
+        {
+            Debug.Assert(!context.IsServer, "GetIdentity: Server is not supported");
+
+            string name = context.Spn;
+            string protocol = context.ProtocolName;
+
+            return new GenericIdentity(name, protocol);
+
+        }
+
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
+        {
+            return NegotiationInfoClass.Kerberos;
+        }
+
+        internal static int QueryMaxTokenSize(string package)
+        {
+            // This value is not used on Unix
+            return 0;
+        }
+
+        internal static string QueryContextClientSpecifiedSpn(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static SafeFreeCredentials AcquireDefaultCredential(string package, bool isServer)
+        {
+            return AcquireCredentialsHandle(package, isServer, new NetworkCredential(string.Empty, string.Empty, string.Empty));
+        }
+
+        internal static SafeFreeCredentials AcquireCredentialsHandle(string package, bool isServer, NetworkCredential credential)
+        {
+            if (isServer)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_server_not_supported);
+            }
+
+            SafeFreeCredentials outCredential;
+            bool isNtlm = string.Equals(package, NegotiationInfoClass.NTLM);
+
+            if (isNtlm)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_ntlm_not_supported);
+            }
```

Comment:
Once we are able to figure out how the fallback works based on our prototyping, we will have to change code here and other places (similar to #5770). This will be part of a future PR so for this PR, the scope is that NTLM is not supported.